src/main/scala/impl/group5/NumberOfLoopsAnalysis.scala

```diff
@@ -0,0 +1,128 @@
+package org.tud.sse.metrics
+package impl.group5
+
+import analysis.{ClassFileAnalysis, MetricValue}
+import input.CliParser.OptionMap
+
+import org.opalj.br.ClassFile
+import org.opalj.br.analyses.Project
+
+import java.net.URL
+import scala.collection.mutable.ListBuffer
+import scala.util.Try
+
+/**
+ * Die Klasse NumberOfLoopsAnalysis stellt die Implementierung  der Metrik Number of Loops (Loop) da.
+ * Diese Metrik zählt für jede Methode die Anzahl der Schleifen und gibt sie aus.
+ *
+ * Die Default Einstellung ist, dass nur die Methoden ausgeben werden, in denen sich Loops befinden. Die anderen Methoden in denen Loops 0 ist werden weggelassen
+ *
+ * Optional CLI argument:
+ *  - --only-class Es wird nur die Anzahl an Loops in einer Klasse angezeigt und nicht für jede Methode
+ *  - --out-all-loops Es werden alle Methoden ausgeben, auch die mit Loop 0.
+ *  - --no-class Es wird nicht die gesamte Zahl der Loops einer Klasse ausgeben
+ *  - --only-class-with-loops Gibt nur Klassen mit Loops aus.
+ */
+
```

Comment:
Die optionalen Parameter tragen verwirrende Bezeichnungen und sind für den Benutzer mißverständlich. Daher schlage ich folgende Neustrukturierung vor:

--allMethods true/FALSE alle Methoden auswerten, auch solche ohne Schleifen (default: FALSE)
--summary (keine Optionen) Summe aller Loops in der Klasse an das Ende der anderen Ergebnisse anhängen
--onlySummary (keine Optionen) Nur die Anzahl aller Loops in der Klasse, überstimmt alle obigen Optionen
--ignoreZeroLoopClasses TRUE/false Klassen ohne Schleifen werden übergangen (default: TRUE)

Reply:
:smiley::thumbsup: